### PR TITLE
Add a Lilac dataset reader.

### PR DESCRIPTION
**Pull request by nsthorat**
# Description

Adds a Lilac reader .

[Lilac](https://lilacml.com/) is an open-source product that helps you analyze, enrich, and clean unstructured data with AI.

It can be used to analyze, clean, structure, and label data that can be used in downstream LlamaIndex and LangChain applications.

## Type of Change

Please delete options that are not relevant.

- [X] New Loader/Tool

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] I stared at the code and made sure it makes sense

I also just tested from python:

IMG_1917

# Suggested Checklist:

- [X] I have added a library.json file if a new loader/tool was added
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_hub/lilac_reader/base.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+              environment variable.
+            text_path: The path to the text field in the dataset. If not defined, uses 'text'.
+            columns (Optional[List[ColumnId]]): The columns to load from the dataset. If not defined, loads all columns.
+            dataset (str): The dataset to load. Should be formatted like {namespace}/{dataset_name}.
+            filters (Optional[Filter]): A filter to apply to the dataset before loading into documents. Useful to filter
+              for labeled data.
+
+        """
+
+        import lilac as ll
```
Comment (EmanuelCampos) [inline llama_hub/lilac_reader/base.py:39]: can you raise an ImportError if the package is not found?

File: poetry.lock
Comment (EmanuelCampos) [inline poetry.lock:1]: remove poetry.lock please

**Review by EmanuelCampos — COMMENTED** [review]
I did a few comments

can you make sure to run  `make format; make lint`?

thanks for the contribution

**Comment by nsthorat** [comment]
Thanks @EmanuelCampos, all are done :)

File: llama_hub/lilac_reader/base.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    Lilac dataset reader
+    """
+
+    def load_data(
+        self,
+        dataset: str,
+        text_path: "Path" = "text",
+        doc_id_path: Optional["Path"] = "doc_id",
+        columns: Optional[List["ColumnId"]] = None,
+        filters: Optional[list["FilterLike"]] = None,
```
Comment (EmanuelCampos) [inline llama_hub/lilac_reader/base.py:23]: line 23: should use typing List instead of list

File: llama_hub/lilac_reader/base.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                    leaf_item = item[path[0]]
+                    if isinstance(leaf_item, dict):
+                        del item[path[0]][ll.VALUE_KEY]
+                    else:
+                        del item[path[0]]
+                return
+            else:
+                _remove_item_path(item[path[0]], path[1:])
+
+        documents: list[Document] = []
```
Comment (EmanuelCampos) [inline llama_hub/lilac_reader/base.py:95]: line 95: should use typing List instead of list

**Review by EmanuelCampos — COMMENTED** [review]
Almost there! did two comments, small things :)

**Comment by EmanuelCampos** [comment]
bump @nsthorat

**Comment by EmanuelCampos** [comment]
quick bump @nsthorat 

if you can make the PR editable by maintainer I can finish it

**Comment by nsthorat** [comment]
Ah, so sorry for dropping this. I made the changes :)

File: llama_hub/lilac_reader/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+"""Init file."""
```
Comment (EmanuelCampos) [inline llama_hub/lilac_reader/__init__.py:1]: just a small change 

```py
from llama_hub.lilac_reader.base import LilacReader

__all__ = [
    "LilacReader",
]
```

**Review by EmanuelCampos — COMMENTED** [review]
Awesome, just a nit then we're good to merge

File: llama_hub/lilac_reader/__init__.py (hunk 4 above)
Comment (nsthorat) [reply llama_hub/lilac_reader/__init__.py:1]: Done!

**Comment by EmanuelCampos** [comment]
@nsthorat the linting broken, we're almost there haha

**Comment by nsthorat** [comment]
haha, done :)

**Review by EmanuelCampos — APPROVED** [review]

**Pull request merged**